internal/pkg/instrumentation/probe/interfaces.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package probe provides instrumentation probe types and definitions.
+package probe
+
+import (
+	"log/slog"
+)
+
+// BaseProbe is the most basic type of Probe. It must support configuration,
+// loading and attaching eBPF programs, running, and closing.
+// All Probes must implement this Base functionality.
+type BaseProbe interface {
```

Comment:
When do we expect this interface to be satisfied but not the others?

This seems like an over generalization for a project designed to produce auto-instrumentation. Shouldn't every probe produce telemetry? When they don't why would they use this project?

Reply:
Every Probe should enable telemetry, but not every Probe will actually produce events. We've already talked about having Probes that don't report telemetry starting in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1105#issuecomment-2353697087 (ie, `ReportingProbe` and `JobProbe` / `SyncProbe` and `AsyncProbe`) when we were looking at setting the boolean for the auto sdk.

We are also now looking at a set of 3 Probes for @grcevski's new approach to context propagation: one to track socket connections, one to allocate header space, and one to insert the new header. In this case, only the 3rd probe actually handles a telemetry event (as I understand it).

So, I don't think we should assume that each Probe object is necessarily going to be emitting telemetry, or we design a way to couple enabling eBPF programs to a telemetry-producing Probe. There's pros and cons to that as well